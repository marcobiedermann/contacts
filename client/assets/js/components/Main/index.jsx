import PropTypes from 'prop-types';
import React from 'react';

const Main = ({ children }) => (
  <main className="main">
    {children}
  </main>
);

Main.propTypes = {
  children: PropTypes.node,
};

Main.defaultProps = {
  children: null,
};

export default Main;
