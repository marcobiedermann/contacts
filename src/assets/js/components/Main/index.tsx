import React from 'react';
import styles from './style.module.css';

export interface MainProps {
  children: React.ReactNode;
}

function Main(props: MainProps): JSX.Element {
  return <main className={styles.main} {...props} />;
}

export default Main;
