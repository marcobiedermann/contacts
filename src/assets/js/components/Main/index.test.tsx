import { render } from '@testing-library/react';
import React from 'react';
import Main from '.';

describe('main component', () => {
  it('renders correctly', () => {
    expect.assertions(1);

    const { asFragment } = render(<Main />);

    expect(asFragment).toMatchSnapshot();
  });
});
