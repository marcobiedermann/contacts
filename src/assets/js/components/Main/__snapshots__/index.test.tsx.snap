// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main component renders correctly 1`] = `[Function]`;
